pkg/controller/disruption/disruption.go

```diff
@@ -748,15 +748,14 @@ func (dc *DisruptionController) syncStalePodDisruption(ctx context.Context, key
 		return nil
 	}
 
-	newStatus := pod.Status.DeepCopy()
-	updated := apipod.UpdatePodCondition(newStatus, &v1.PodCondition{
-		Type:   v1.AlphaNoCompatGuaranteeDisruptionTarget,
-		Status: v1.ConditionFalse,
-	})
-	if !updated {
-		return nil
-	}
-	if _, _, _, err := utilpod.PatchPodStatus(ctx, dc.kubeClient, namespace, name, pod.UID, pod.Status, *newStatus); err != nil {
+	podApply := corev1apply.Pod(pod.Name, pod.Namespace).WithStatus(corev1apply.PodStatus())
```

Comment:
/hold

Reply:
Why?

Reply:
Just this call, not the others in this PR.

This worker sets the condition to false if the condition has been true for a while and the object doesn't have a deletion timestamp. If, right at the time when this clean is running, something else refreshes the condition, then it means that there is an imminent deletion coming up, and we want to preserve the condition in that case.

Reply:
OK -- if you do that you have to be prepared to retry in the case of irrelevant changes.

An alternative might be to just not force conflicts.

Reply:
We need to force, because the condition is definitely managed by another manager at this point (like kube-scheduler)

Reply:
I see. That's too bad.

Reply:
> Just this call, not the others in this PR.
> 
> This worker sets the condition to false if the condition has been true for a while and the object doesn't have a deletion timestamp. If, right at the time when this clean is running, something else refreshes the condition, then it means that there is an imminent deletion coming up, and we want to preserve the condition in that case.

The scenario you describe assumes that we have 2min passed already without deleting so that the disruption controller attempts to change the `DisruptionTarget`'s `status` to `False`. Then, the suggested change would make a difference only if the `DisruptionTarget` condition, is added by another controller in the time window between `Get` and `Apply` requests (from the disruption controller point of view - within the `syncStalePodDisruption` function). 

Even if we implement the suggested change, the race situation may result in disruption controller setting the `Status` to `False` if the other controller updates the condition just before disruption controller does the `Get` call.

Also, I don't see how to cover it with tests - in integration tests we could shift the clock by 2min and try to add the `DisruptionTarget` condition roughly at the same time as disruption controller attempts to change it to `False`, but even then the expected outcome might be `True` or `False`, depending on the relative timing.

To me it seems like an unnecessary complication which is hard to cover with tests, but I'm ok with making the change (it is one-liner) if this is the consensus @alculquicondor @apelisse @lavalamp ?

Reply:
if you include the resourceversion in the patch, then the API update will fail if there was a change.

It might be a one-line change. Just make sure whether we need to retry here. I guess the just returning an error would cause a retry, right?

Reply:
Added the resource version. 

Also, tested manually that the patch is rejected if there is a pod update in the meanwhile (which could be an update to the stale pod condition). Also tested that if the operation fails it gets retried on the next iteration of the `syncStalePodDisruption`.